pkg/fanal/secret/scanner.go

```diff
@@ -435,8 +436,22 @@ func (s *Scanner) Scan(args ScanArgs) types.Secret {
 		}
 	}
 
-	for _, match := range matched {
-		findings = append(findings, toFinding(match.Rule, match.Location, censored))
+	if args.Binary {
+		for _, match := range matched {
+			findings = append(findings, types.SecretFinding{
+				RuleID:    match.Rule.ID,
+				Category:  match.Rule.Category,
+				Severity:  lo.Ternary(match.Rule.Severity == "", "UNKNOWN", match.Rule.Severity),
+				Title:     match.Rule.Title,
+				Match:     fmt.Sprintf("Binary file %q matches a rule %q", args.FilePath, match.Rule.Title),
+				StartLine: 1,
+				EndLine:   1,
+			})
+		}
+	} else {
+		for _, match := range matched {
+			findings = append(findings, toFinding(match.Rule, match.Location, censored))
+		}
```

Comment:
I sugest to use `toFinding` function.
something like this:
```suggestion
		for _, match := range matched {
		finding := toFinding(match.Rule, match.Location, censored)
		// These fields will be unreadable for binaries,
		// Therefore overwrite them.		
		if args.Binary {
			finding.Match = fmt.Sprintf("Binary file %q matches a rule %q", args.FilePath, match.Rule.Title)
			finding.Code = types.Code{}
		}
		findings = append(findings, finding)
	}
```
It might make sense to add flag to `toFinding` function to skip `findLocation`.

wdyt?

Reply:
sure! it's a better way!